src/twisted/internet/test/test_endpoints.py

```diff
@@ -3259,6 +3259,29 @@ def test_sslNoTrailingNewlinePem(self):
         ctx = server._sslContextFactory.getContext()
         self.assertIsInstance(ctx, ContextType)
 
+    @skipIf(skipSSL, skipSSLReason)
+    def test_sslWithCustomCipher(self):
+        """
+        Test adding custom cipher to the serverFromString function.
+        """
+        reactor = object()
+        server = (
+            endpoints.serverFromString(
+                reactor,
+                "ssl:1234:backlog=12:privateKey=%s:"
+                "certKey=%s:sslmethod=TLSv1_2_METHOD:interface=10.0.0.1:cipher=ECDHE-ECDSA-AES128-GCM-SHA256,ECDHE-ECDSA-CHACHA20-POLY1305,ECDHE-RSA-AES128-GCM-SHA256"
+                % (escapedPEMPathName, escapedPEMPathName),
+            ),
+        )
+        self.assertIsInstance(server, endpoints.SSL4ServerEndpoint)
+        self.assertIs(server._reactor, reactor)
+        self.assertEqual(server._port, 1234)
+        self.assertEqual(server._backlog, 12)
+        self.assertEqual(server._interface, "10.0.0.1")
+        self.assertEqual(server._sslContextFactory.method, TLSv1_2_METHOD)
```

Comment:
ar these assertions needed for this test?

I guess that here we should assert that the ciphers were set...

I don't know how to get the configured cipher list for an SSL context :(

-------

For my code, to test it, I do an actual handshake...

but I feel that that it's too much here.